### PR TITLE
[Fix] Fix the error of one hot when ignore_index is bigger than class nums

### LINKED ISSUE #2270: [Bug] Dice loss bug
环境：aistudio tesla v100
paddlepaddle=2.3.0
paddleseg=2.5.0

bug：dice loss 前向传播时，label转换为one-hot编码前，并未对ignore_index进行处理，当ignore_index值大于num_classes（比如ignore index为255，类别为19）时，报错cuda 719。原因为one-hot转换错误。 
代码链接：https://github.com/PaddlePaddle/PaddleSeg/blob/35a4c4d229df2d4a5ca724ad442bf5e0f75b4823/paddleseg/models/losses/dice_loss.py#L46

可将mask部分放到one-hot之前，然后将ignore_index赋值一个小于num_classes的值：
```python
 def forward(self, logits, labels):
        num_class = logits.shape[1]
        if self.weight is not None:
            assert num_class == len(self.weight), \
                "The lenght of weight should be euqal to the num class"
        if logits.shape != labels.shape:
            labels = labels.unsqueeze(axis=1)
            labels = F.interpolate(labels, size=logits.shape[2:], mode='nearest')
            labels = labels.squeeze(axis=1)
        logits = F.softmax(logits, axis=1)
        mask = labels != self.ignore_index
        mask = paddle.cast(paddle.unsqueeze(mask, 1), 'float32')
        labels[labels == self.ignore_index] = 0
        labels_one_hot = F.one_hot(labels, num_class)
        labels_one_hot = paddle.transpose(labels_one_hot, [0, 3, 1, 2])
        
        dice_loss = 0.0
        for i in range(num_class):
            dice_loss_i = dice_loss_helper(logits[:, i], labels_one_hot[:, i],
                                           mask, self.smooth, self.eps)
            if self.weight is not None:
                dice_loss_i *= self.weight[i]
            dice_loss += dice_loss_i
        dice_loss = dice_loss / num_class

        return dice_loss
```

### PR DESCRIPTION
**Pull request by juncaipeng**
Fix #2270

### CODE REVIEW

**Review by shiyutang — APPROVED** [review]
LGTM
